### PR TITLE
Refactor iasmdmd.  Convert multiple optional pointer parameters to slices

### PR DESCRIPTION
**Pull request by benjones**
Probably easiest to see commit by commit.  The last commit got a bit bigger than it probably should have.

### CODE REVIEW

File: src/dmd/iasmdmd.d [hunk 1]
```diff
@@ -178,23 +179,23 @@ version (none) // don't use bReturnax anymore, and will fail anyway if we use re
...
                         asm_token();
-                        asm_cond_exp(opnd3);
-                        o3 = &opnd3;
+                        asm_cond_exp(opnds[2]);
+                        nOps = 3;
                         if (asmstate.tokValue == TOK.comma)
                         {
                             asm_token();
-                            asm_cond_exp(opnd4);
-                            o4 = &opnd4;
```
Comment (thewilsonator) [inline src/dmd/iasmdmd.d:197]: surely this can be converted into a loop

File: src/dmd/iasmdmd.d [hunk 2]
```diff
@@ -740,34 +699,19 @@ PTRNTAB asm_classify(OP *pop, OPND *popnd1, OPND *popnd2,
 
     void TYPE_SIZE_ERROR()
     {
-        if (popnd1 && ASM_GET_aopty(popnd1.usFlags) != _reg)
+        foreach (i, ref opnd; opnds)
         {
-            opflags1 = popnd1.usFlags |= _anysize;
-            if (asmstate.ucItype == ITjump)
+            if (ASM_GET_aopty(opnd.usFlags) != _reg)
```
Comment (thewilsonator) [inline src/dmd/iasmdmd.d:704]:
```
if (ASM_GET_aopty(opnd.usFlags) == _reg)
    continue;
```

File: src/dmd/iasmdmd.d [hunk 3]
```diff
@@ -740,34 +699,19 @@ PTRNTAB asm_classify(OP *pop, OPND *popnd1, OPND *popnd2,
...
             {
-                if (bRetry && popnd1.s && !popnd1.s.isLabel())
+                opflags[i] = opnd.usFlags |= _anysize;
+                if(asmstate.ucItype == ITjump)
```
Comment (thewilsonator) [inline src/dmd/iasmdmd.d:707]: ditto

File: src/dmd/iasmdmd.d [hunk 4]
```diff
@@ -2452,7 +2342,7 @@ void asm_make_modrm_byte(
         ubyte *puchOpcode, uint *pusIdx,
         code *pc,
         uint usFlags,
-        OPND *popnd, OPND *popnd2)
+        scope OPND[] opnds)
```
Comment (thewilsonator) [inline src/dmd/iasmdmd.d:2345]: the length is always 2, you could do `scope ref OPND[2] opnds`

File: src/dmd/iasmdmd.d [hunk 5]
```diff
@@ -2924,7 +2814,7 @@ void asm_make_modrm_byte(
 /*******************************
  */
 
-regm_t asm_modify_regs(PTRNTAB ptb, OPND *popnd1, OPND *popnd2)
+regm_t asm_modify_regs(PTRNTAB ptb, OPND[] opnds)
```
Comment (thewilsonator) [inline src/dmd/iasmdmd.d:2817]: `scope`

File: src/dmd/iasmdmd.d (hunk 4 above)
Comment (benjones) [reply src/dmd/iasmdmd.d:2345]: The length is 1 or 2.  In cases where null was pass previously, not a single element slice is passed.

Comment (thewilsonator) [reply src/dmd/iasmdmd.d:2345]: OK.

File: src/dmd/iasmdmd.d (hunk 1 above)
Comment (benjones) [reply src/dmd/iasmdmd.d]: Ha, of course.  I think the one I came up with is equiavalent

**Pull request merged**